src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -38,25 +39,42 @@ public FindCommand parse(String args) throws ParseException {
         }
 
-        if (argMultimap.getValue(PREFIX_NAME).isPresent() && argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+        if (argMultimap.getValue(PREFIX_NAME).isPresent() && argMultimap.getValue(PREFIX_PHONE).isPresent()
+            && argMultimap.getValue(PREFIX_EMAIL).isEmpty() && argMultimap.getValue(PREFIX_TAG).isEmpty()
+            && argMultimap.getValue(PREFIX_JOBCODE).isEmpty()) {
             // Name and Phone search
             String name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()).fullName;
             String phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get()).value;
             return new FindCommand(new NamePhonePredicate(name, phone));
-        } else if (argMultimap.getValue(PREFIX_NAME).isPresent() && argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+        } else if (argMultimap.getValue(PREFIX_NAME).isPresent() && argMultimap.getValue(PREFIX_EMAIL).isPresent()
+            && argMultimap.getValue(PREFIX_PHONE).isEmpty() && argMultimap.getValue(PREFIX_TAG).isEmpty()
+            && argMultimap.getValue(PREFIX_JOBCODE).isEmpty()) {
             // Name and Email search
             String name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()).fullName;
             String email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get()).value;
             return new FindCommand(new NameEmailPredicate(name, email));
-        } else if (argMultimap.getValue(PREFIX_JOBCODE).isPresent()) {
+        } else if (argMultimap.getValue(PREFIX_JOBCODE).isPresent() && argMultimap.getValue(PREFIX_TAG).isPresent()
+                && argMultimap.getValue(PREFIX_PHONE).isEmpty() && argMultimap.getValue(PREFIX_NAME).isEmpty()
+                && argMultimap.getValue(PREFIX_EMAIL).isEmpty()) {
+            // Job Code and Tag search
+            String jobCode = ParserUtil.parseJobCode(argMultimap.getValue(PREFIX_JOBCODE).get()).value;
+            String tag = ParserUtil.parseTag(argMultimap.getValue(PREFIX_TAG).get()).tagCode;
+            return new FindCommand(new JobCodeTagPredicate(jobCode, tag));
+        } else if (argMultimap.getValue(PREFIX_JOBCODE).isPresent()
+            && argMultimap.getValue(PREFIX_PHONE).isEmpty() && argMultimap.getValue(PREFIX_TAG).isEmpty()
+            && argMultimap.getValue(PREFIX_NAME).isEmpty() && argMultimap.getValue(PREFIX_EMAIL).isEmpty()) {
             // Job Code search
             String jobCode = ParserUtil.parseJobCode(argMultimap.getValue(PREFIX_JOBCODE).get()).value;
             return new FindCommand(new JobCodePredicate(jobCode));
-        } else if (argMultimap.getValue(PREFIX_TAG).isPresent()) {
+        } else if (argMultimap.getValue(PREFIX_TAG).isPresent()
+            && argMultimap.getValue(PREFIX_PHONE).isEmpty() && argMultimap.getValue(PREFIX_JOBCODE).isEmpty()
+            && argMultimap.getValue(PREFIX_NAME).isEmpty() && argMultimap.getValue(PREFIX_EMAIL).isEmpty()) {
```

Comment:
Good use of guard clauses! Have you thought about refactoring this large condition into smaller helper methods for readability and maintainability? Example: 'isNamePhoneSearch()' could make the logic more self-explanatory.